profile-googlefonts/src/checks/googlefonts/name/license.rs

```diff
@@ -48,78 +53,78 @@ fn license(t: &Testable, _context: &Context) -> CheckFnResult {
         ),
     ]);
 
-    let f = testfont!(t);
     let mut problems = vec![];
     let mut http_warn = false;
+    let mut entry_found: bool = false;
 
-    // FIXME: This should probably come from a TestableCollection, I guess...
-    let license_filename = "LICENSE.txt";
+    #[allow(clippy::unwrap_used)]
+    let license_filename = c
+        .get_file("OFL.txt")
+        .or_else(|| c.get_file("LICENSE.txt"))
+        .ok_or_else(|| CheckError::skip("license-file-missing", "A license file was not found."))?
+        .filename
+        .file_name()
+        .unwrap()
+        .to_string_lossy();
 
-    let placeholder = placeholder_licensing_text[license_filename];
-    let mut entry_found: bool = false;
-    let name = f.font().name()?;
-    let mut license_description = "".to_string();
-    for record in name.name_record().iter() {
-        let string = record.string(name.string_data())?;
-        if record.name_id() == StringId::LICENSE_DESCRIPTION {
-            license_description = string.to_string();
-            entry_found = true;
-            if license_description.contains("http://") {
-                problems.push(Status::warn(
-                    "http-in-description",
-                    &format!(
-                        "Please consider using HTTPS URLs at \
-                              name table entry \
-                              [plat={}, enc={}, name={}]",
-                        record.platform_id(),
-                        record.encoding_id(),
-                        record.name_id()
-                    ),
-                ));
-                license_description = license_description.replace("http://", "https://");
-                http_warn = true;
+    let placeholder = placeholder_licensing_text[&license_filename.deref()];
```

Comment:
In general I am not a fan on `[]` notation, as if the element isn't found, it will panic. But in this case, it's OK as the cases are known at compile time.

So instead, here is a small style issue: this reads a lot like Python converted to Rust, not Rust. In Python we would have no problems creating a dictionary to do lookups. But there is no need to create and allocate a dictionary here, particularly for a static lookup table - I would consider using a `match` statement instead.